mssmt/compacted_tree.go

```diff
@@ -392,3 +392,166 @@ func (t *CompactedTree) MerkleProof(ctx context.Context, key [hashSize]byte) (
 
 	return NewProof(proof), nil
 }
+
+// collectLeaves recursively traverses the compacted tree represented by the
+// given node and collects all non-empty leaf nodes along with their keys.
+func collectLeaves(ctx context.Context, tx TreeStoreViewTx, node Node) (
```

Comment:
I don't think this function is used. Is it included intentionally?

Reply:
Yep not needed, can remove.